CMakeLists.txt

```diff
@@ -322,27 +335,29 @@ target_link_libraries(osrm-extract osrm_extract ${Boost_LIBRARIES})
 target_link_libraries(osrm-contract osrm_contract ${Boost_LIBRARIES})
 target_link_libraries(osrm-routed osrm ${Boost_LIBRARIES} ${OPTIONAL_SOCKET_LIBS} ${ZLIB_LIBRARY})
 
-set(EXTRACTOR_LIBRARIES
-    ${BZIP2_LIBRARIES}
-    ${Boost_LIBRARIES}
-    ${CMAKE_THREAD_LIBS_INIT}
-    ${EXPAT_LIBRARIES}
-    ${LUABIND_LIBRARY}
-    ${USED_LUA_LIBRARIES}
-    ${OSMIUM_LIBRARIES}
-    ${STXXL_LIBRARY}
-    ${TBB_LIBRARIES}
-    ${ZLIB_LIBRARY}
-    ${MAYBE_COVERAGE_LIBRARIES})
-set(CONTRACTOR_LIBRARIES
-    ${Boost_LIBRARIES}
-    ${CMAKE_THREAD_LIBS_INIT}
-    ${LUABIND_LIBRARY}
-    ${USED_LUA_LIBRARIES}
-    ${STXXL_LIBRARY}
-    ${TBB_LIBRARIES}
-    ${MAYBE_RT_LIBRARY}
-    ${MAYBE_COVERAGE_LIBRARIES})
+if(NOT ROUTED_STANDALONE)
```

Comment:
If we really want to support this, we need to split the CMakeList not add a bunch of switches all over the place.

@daniel-j-h you did this for the `unit_tests` what are you experiences w.r.t to complexity here?